frontend/src/router.js

```diff
@@ -1022,13 +1022,17 @@ const router = createRouter({
 
 router.beforeEach((to, _from, next) => {
   if (to.meta.requiresAuth) {
+    const appStore = useAppStore();
     const authStore = useAuthStore();
     authStore
       .getJwtToken()
-      .then(() => {
+      .then(async () => {
         if (!authStore.isAuthenticated) {
           next('/token-expired');
         } else {
+          if (isEmpty(appStore.programYearList)) {
```

Comment:
1. Can we add a comment explaining why we're doing this?
2. We should probably just check lookinfo itself. Yes, it's programYearList that fails but the wider issue is that lookupInfo hasn't loaded